### ISSUE TITLE
`install` `signin` & `signout` fail on High Sierra 10.13.3

### ISSUE THREAD

**Issue by stalinkay** [mas-cli/mas#129]
I'm able to sign in or sign out using `mas signin xxxxxx@xxxxx.xxx; mas signout`. `mas install xxxxxx` stalls doesn't do anything. Please look into this.

**Comment by phatblat** [comment]
This sounds like #78. Have you tried `mas signin --dialog`?

**Comment by mikumi** [comment]
I have the same issue, signing out does not seem to have any effect:

```
MKs-MBP2017.local  ~  mas signout                            
MKs-MBP2017.local  ~  mas signin --dialog abc@xyz.com
Error: Already signed in
MKs-MBP2017.local  ~ 
```

**Comment by anxsec** [comment]
The same issue as above.
```
➜  ~ mas signout
➜  ~ mas signin --dialog xxx@xxx.com
Error: Already signed in
```

**Comment by norman-chan** [comment]
Running macOS 10.13.3 and here's what I get using mas signin --dialog:

Note: Also experiencing signout issue where it does not sign out of App Store.

`mas signin --dialog xxx@xxx.om

==> Signing in to Apple ID: xxx@xxx.com
2018-03-27 10:15:25.804 mas[6351:4600309] -[__NSXPCInterfaceProxy_ISAccountService signInWithContext:replyBlock:]: unrecognized selector sent to instance 0x7ffc8890a3e0
2018-03-27 10:15:25.805 mas[6351:4600309] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSXPCInterfaceProxy_ISAccountService signInWithContext:replyBlock:]: unrecognized selector sent to instance 0x7ffc8890a3e0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff2b8c533b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff52609942 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff2b95c5e4 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff2b83cf03 ___forwarding___ + 1443
	4   CoreFoundation                      0x00007fff2b83c8d8 _CF_forwarding_prep_0 + 120
	5   mas                                 0x000000010b0cc452 mas + 201810
	6   mas                                 0x000000010b0e9b70 mas + 322416
	7   mas                                 0x000000010b0b51e4 mas + 106980
	8   mas                                 0x000000010b0a16fd mas + 26365
	9   mas                                 0x000000010b0a1590 mas + 26000
	10  mas                                 0x000000010b0b3276 mas + 98934
	11  mas                                 0x000000010b09e7ef mas + 14319
	12  libdyld.dylib                       0x00007fff531f6115 start + 1
	13  ???                                 0x0000000000000004 0x0 + 4
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6`

**Comment by HeminWon** [comment]
me too

**Comment by SylvainCecchetto** [comment]
Same issue for me:
```
$ mas signout
$ mas signin ********@****
   Error: Already signed in
```

**Comment by dmlemos** [comment]
I am also having this issue. Can only signin with sudo, otherwise nothing works.

**Comment by RishabhTayal** [comment]
same issue for me.

**Comment by phatblat** [comment]
It's clear from all your feedback that on High Sierra:

1. `mas signout` does nothing
1. `mas signin ...` crashes

However, it's not clear from the comments what the issue with `mas install` is. Perhaps it was just included because the original report included it in the chain of commands?

**Comment by phatblat** [comment]
Dup of #107

**Comment by phatblat** [comment]
Quick update: I've got signout working now. Signin is giving me some trouble as the APIs have changed and I've got to do some digging to see how we can authenticate now.

**Comment by thaiphv** [comment]
Same for me
```bash
$ mas signin xxxxx@xxxxxxx.com            
==> Signing in to Apple ID: xxxx@xxxxxxx.com
Password: 
2018-05-15 21:15:59.833 mas[8499:170862] -[__NSXPCInterfaceProxy_ISAccountService signInWithContext:replyBlock:]: unrecognized selector sent to instance 0x7fc90562a4e0
2018-05-15 21:15:59.836 mas[8499:170862] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSXPCInterfaceProxy_ISAccountService signInWithContext:replyBlock:]: unrecognized selector sent to instance 0x7fc90562a4e0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff540926bb __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff7b7a9942 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff541299e4 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff5400a183 ___forwarding___ + 1443
	4   CoreFoundation                      0x00007fff54009b58 _CF_forwarding_prep_0 + 120
	5   mas                                 0x000000010caea462 _T0So14ISStoreAccountC3masE6signInABSSSg8username_AE8passwordSb12systemDialogtKFZTf4ggnd_n + 1026
	6   mas                                 0x000000010cb07b80 _T03mas13SignInCommandV3runAA6ResultOyytAA8MASErrorOGAA0bC7OptionsVFTf4gXd_nTf4xnn_n + 592
	7   mas                                 0x000000010cad31f4 _T03mas14CommandWrapperVACyxGqd__c11ClientErrorQyd__RszAA0B8ProtocolRd__7Options_AEQYd__AFRSlu33_E513F8A29DAA43E418079999EEBED6A7LlfcAA6ResultOyytAA010CommandantE0OyxGGAA14ArgumentParserCcfU_AA8MASErrorO_AA06SignInB0VTg5Tf4gg_n + 340
	8   mas                                 0x000000010cabf70d _T03mas14CommandWrapperVACyxGqd__c11ClientErrorQyd__RszAA0B8ProtocolRd__7Options_AEQYd__AFRSlu33_E513F8A29DAA43E418079999EEBED6A7LlfcAA6ResultOyytAA010CommandantE0OyxGGAA14ArgumentParserCcfU_AA8MASErrorO_AA04InfoB0VTG5TATm + 29
	9   mas                                 0x000000010cabf5a0 _T03mas14CommandWrapperVACyxGqd__c11ClientErrorQyd__RszAA0B8ProtocolRd__7Options_AEQYd__AFRSlu33_E513F8A29DAA43E418079999EEBED6A7LlfcAA6ResultOyytAA010CommandantE0OyxGGAA14ArgumentParserCcfU_AA8MASErrorO_AA06SignInB0VTG5TA + 16
	10  mas                                 0x000000010cad1286 _T03mas15CommandRegistryC4mains5NeverOSaySSG9arguments_SS11defaultVerbyxc12errorHandlertFAA8MASErrorO_Tg508_T03mas8K42OIxx_ACIxi_TR011_T03masyAA8B5OcfU_Tf3npf_nTf1nncn_nTf4nxn_n + 518
	11  mas                                 0x000000010cabc7ff main + 5391
	12  libdyld.dylib                       0x00007fff7c3c0015 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6
```

**Comment by cha0ran** [comment]
macOS 10.13.4 (17E202)

1. `mas signout` it's not working 
2. `mas signin xxxxx@xx.com` return Error: Already signed in

**Comment by valdotdev** [comment]
What happened? on `10.13.4`

```
$ mas signin ????.????@?????.com
==> Signing in to Apple ID: ????.????@?????.com
Password:
2018-06-05 23:53:34.651 mas[2938:42211] -[__NSXPCInterfaceProxy_ISAccountService signInWithContext:replyBlock:]: unrecognized selector sent to instance 0x7f904bd2eb20
2018-06-05 23:53:34.653 mas[2938:42211] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSXPCInterfaceProxy_ISAccountService signInWithContext:replyBlock:]: unrecognized selector sent to instance 0x7f904bd2eb20'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff4e0df32b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff75755c76 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff4e177e04 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff4e055870 ___forwarding___ + 1456
	4   CoreFoundation                      0x00007fff4e055238 _CF_forwarding_prep_0 + 120
	5   mas                                 0x00000001059da462 _T0So14ISStoreAccountC3masE6signInABSSSg8username_AE8passwordSb12systemDialogtKFZTf4ggnd_n + 1026
	6   mas                                 0x00000001059f7b80 _T03mas13SignInCommandV3runAA6ResultOyytAA8MASErrorOGAA0bC7OptionsVFTf4gXd_nTf4xnn_n + 592
	7   mas                                 0x00000001059c31f4 _T03mas14CommandWrapperVACyxGqd__c11ClientErrorQyd__RszAA0B8ProtocolRd__7Options_AEQYd__AFRSlu33_E513F8A29DAA43E418079999EEBED6A7LlfcAA6ResultOyytAA010CommandantE0OyxGGAA14ArgumentParserCcfU_AA8MASErrorO_AA06SignInB0VTg5Tf4gg_n + 340
	8   mas                                 0x00000001059af70d _T03mas14CommandWrapperVACyxGqd__c11ClientErrorQyd__RszAA0B8ProtocolRd__7Options_AEQYd__AFRSlu33_E513F8A29DAA43E418079999EEBED6A7LlfcAA6ResultOyytAA010CommandantE0OyxGGAA14ArgumentParserCcfU_AA8MASErrorO_AA04InfoB0VTG5TATm + 29
	9   mas                                 0x00000001059af5a0 _T03mas14CommandWrapperVACyxGqd__c11ClientErrorQyd__RszAA0B8ProtocolRd__7Options_AEQYd__AFRSlu33_E513F8A29DAA43E418079999EEBED6A7LlfcAA6ResultOyytAA010CommandantE0OyxGGAA14ArgumentParserCcfU_AA8MASErrorO_AA06SignInB0VTG5TA + 16
	10  mas                                 0x00000001059c1286 _T03mas15CommandRegistryC4mains5NeverOSaySSG9arguments_SS11defaultVerbyxc12errorHandlertFAA8MASErrorO_Tg508_T03mas8K42OIxx_ACIxi_TR011_T03masyAA8B5OcfU_Tf3npf_nTf1nncn_nTf4nxn_n + 518
	11  mas                                 0x00000001059ac7ff main + 5391
	12  libdyld.dylib                       0x00007fff7636f015 start + 1
	13  ???                                 0x0000000000000003 0x0 + 3
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6
```

**Comment by jonathan-automox** [comment]
Is this project still active?  Anything being done to support latest MacOS?

**Comment by MoOx** [comment]
FYI the error happens with 10.14 too (Mojave).

**Comment by phatblat** [comment]
Yep, this project is still active, however this issue has been insurmountable due to the `signin` command relying on a private API that was removed in High Sierra. The only option at the moment is to sign into the Mac App Store GUI manually.

**Comment by nalexander50** [comment]
I tried logging into the Mac App Store manually, but `mas` is still reporting that I am not signed in.

**Comment by MoOx** [comment]
Same for me on Mojave. I am signed for sure, but `mas` don't get it. Probably another API change... :/

**Comment by phatblat** [comment]
`signout` command fixed and `signin` crasher suppressed by #162
